mathics/core/evaluation.py

```diff
@@ -117,10 +149,11 @@ def run_with_timeout_and_stack(request, timeout, evaluation):
     thread.join(timeout)
     if thread.is_alive():
         evaluation.timeout = True
-        while thread.is_alive():
-            pass
-        evaluation.timeout = False
+        time.sleep(0.01)
```

Comment:
When I see a `time.sleep(x)` I always feel that something is fishy. This kind of think should only be used as a _last_ resort. 

Will think about this some more.

Reply:
I wonder if 0.01 is related to `$TimeUnit` which we don't have defined.

Reply:
Is related with the overhead I see in my computer. This is something to refine.

Reply:
Regarding *sleep*, is what Pause does in WL. Is there a better way to do it?

Reply:
I suspect if you were to do this  with the Python's [multiprocessing](https://docs.python.org/3/library/multiprocessing.html) module also mentioned in https://www.geeksforgeeks.org/python-different-ways-to-kill-a-thread/  it would work in Windows, PyPy, and pyston without the sleeps and with more reliable (non-GIL) threads and therefore also faster. 

The only downside is that it requires a CPU with more than one core, but that is pretty much all CPU's except for little ARM matchbox processors. But those won't run this anyway.

Reply:
I have to check it. What decides me to use threads instead the process is that we are working over the same ``evaluation.definitions`` object. So, it makes sense to use a what that allows sharing memory.  

On the other hand, at some point the multithread module should support this "feature" in the other platforms, isn't it?

Reply:
The docs say it runs on Windows and GNU/Linux. Given that I can't imagine why it wouldn't also be supported on on MacOS. 

And I think you will find it is faster. Please give it a try.

Reply:
> What decides me to use threads instead the process is that we are working over the same evaluation.definitions object

This might be true if there weren't the Global Interpreter Lock (GIL) that slows things down here.